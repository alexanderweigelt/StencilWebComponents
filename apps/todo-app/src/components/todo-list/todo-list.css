ul {
  list-style: none;
  margin: 0;
  padding: 0;
}

li {
  background: #cccccc;
  margin: 0.5rem 0;
  padding: 0.5rem 1rem;
  border-radius: 0.25rem;
}

ui-button {
  margin-right: 0.5rem;
}
