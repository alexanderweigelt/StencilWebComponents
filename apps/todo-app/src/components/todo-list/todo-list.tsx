import { Component, Host, h } from '@stencil/core';

@Component({
  tag: 'todo-list',
  shadow: true,
})
export class TodoList {
  render() {
    return (
      <Host>
        <slot></slot>
      </Host>
    );
  }
}
