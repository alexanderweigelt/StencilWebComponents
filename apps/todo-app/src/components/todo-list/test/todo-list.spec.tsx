import { newSpecPage } from '@stencil/core/testing';
import { TodoList } from '../todo-list';
import { h } from '@stencil/core';

describe('ToDoList Component', () => {
  const mockTodos = [
    { id: 1, text: 'Learn Stencil', dueDate: '20.12.2024, 17:40:45', completed: false },
    { id: 2, text: 'Write Tests', dueDate: '24.12.2024, 18:45:48', completed: true },
  ];

  it('matches the snapshot with todos', async () => {
    const page = await newSpecPage({
      components: [],
      template: () => <TodoList todos={mockTodos} toggleTodo={() => {
      }} />,
    });

    expect(page.root).toMatchSnapshot();
  });

  it('matches the snapshot with an empty todos array', async () => {
    const page = await newSpecPage({
      components: [],
      template: () => <TodoList todos={[]} toggleTodo={() => {
      }} />,
    });

    expect(page.root).toMatchSnapshot();
  });
});
