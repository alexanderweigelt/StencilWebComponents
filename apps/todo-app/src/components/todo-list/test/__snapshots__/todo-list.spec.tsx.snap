// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList Component matches the snapshot with an empty todos array 1`] = `<ul class="todo-list"></ul>`;

exports[`ToDoList Component matches the snapshot with todos 1`] = `
<ul class="todo-list">
  <li class="todo-item">
    <ui-button>
      open
    </ui-button>
    <ui-text>
      Learn Stencil
    </ui-text>
  </li>
  <li class="completed todo-item">
    <ui-button>
      complete
    </ui-button>
    <ui-text>
      Write Tests
    </ui-text>
  </li>
</ul>
`;
